activerecord/lib/active_record/relation/predicate_builder.rb

```diff
@@ -107,25 +107,33 @@ def create_binds_for_hash(attributes)
               last = Arel::Nodes::BindParam.new
             end
 
-            result[column_name] = RangeHandler::RangeWithBinds.new(first, last, value.exclude_end?)
-          else
-            if can_be_bound?(column_name, value)
-              result[column_name] = Arel::Nodes::BindParam.new
-              binds << build_bind_param(column_name, value)
-            end
+            value = RangeHandler::RangeWithBinds.new(first, last, value.exclude_end?)
+          when can_be_bound?(column_name, value)
+            binds << build_bind_param(column_name, value)
+            value = Arel::Nodes::BindParam.new
           end
 
-          # Find the foreign key when using queries such as:
-          # Post.where(author: author)
-          #
-          # For polymorphic relationships, find the foreign key and type:
-          # PriceEstimate.where(estimate_of: treasure)
-          if table.associated_with?(column_name)
-            result[column_name] = AssociationQueryHandler.value_for(table, column_name, value)
-          end
+          parts << build(table.arel_attribute(column_name), value)
+        end
+
+        [parts, binds]
+      end
+
+      def build_for_association_query(column_name, value)
+        parts, binds = [], []
+        AssociationQueryValue.queries_for(table, column_name, value).each do |query|
+          prts, bvs = build_from_hash(query)
+          parts << prts
```

Comment:
Cannot use `parts.concat`. `binds` should be flatten, but `parts` should be kept `type` and `id` pair.

```ruby
parts = [[:prts1_type, :prts2_id], [:prts2_type, :prts2_id]]
binds = [:bvs1_type, :bvs1_id, :bvs2_type, :bvs2_id]
```